test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceProviderTests.cs

```diff
@@ -963,6 +963,60 @@ public void SavePackageSourcesWithHierarchyClear()
         }
 
         [Fact]
+        public void SavePackageSources_RetainUnavailableDisabledSources()
+        {
+            using (var mockBaseDirectory = TestFilesystemUtility.CreateRandomTestFolder())
+            {
+                // Arrange
+                var configContents =
+                    @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+        <add key=""a"" value=""https://a"" />
+        <add key=""b"" value=""https://b"" />
+    </packageSources>
+    <disabledPackageSources>
+        <add key=""Microsoft and .NET"" value=""true"" />
+        <add key=""b"" value=""true"" />
+    </disabledPackageSources>
+</configuration>
+";
+                File.WriteAllText(Path.Combine(mockBaseDirectory.Path, "NuGet.config"), configContents);
+
+                var rootPath = Path.Combine(mockBaseDirectory.Path, Path.GetRandomFileName());
+
+                var settings = Settings.LoadDefaultSettings(rootPath,
+                    configFileName: null,
+                    machineWideSettings: null,
+                    loadAppDataSettings: false);
+
+                var expectedDisabledSources = settings.GetSettingValues("disabledPackageSources")?.ToList();
```

Comment:
Theoretically, settings does not need to contain the "disabledPackageSources" section at all. In that case, GetSettingValues method returns null. Hence, the use of '?'